File: 1-easy/problems/1582-special-positions-in-a-binary-matrix/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def numSpecial(self, mat):
        """
        :type mat: List[List[int]]
        :rtype: int
        """
        rows, cols = [0]*len(mat), [0]*len(mat[0])
        for i in range(len(rows)):
            for j in range(len(cols)):
                if mat[i][j]:
                    rows[i] += 1
                    cols[j] += 1
        result = 0
        for i in range(len(rows)):
            for j in range(len(cols)):
                if mat[i][j] == rows[i] == cols[j] == 1:
                    result += 1
        return result

File: 1-easy/problems/1582-special-positions-in-a-binary-matrix/solution.ts
function numSpecial(mat: any): boolean | number | string | any {
  """
  :type mat: List[List[int]]
  :rtype: int
  """
  rows, cols = [0]*(mat).length, [0]*(mat[0]).length
  for (i in Array.from({length: (rows}, (_, i) => i).length)) {
      for (j in Array.from({length: (cols}, (_, i) => i).length)) {
          if (mat[i][j]) {
              rows[i] += 1
              cols[j] += 1
  result = 0
  for (i in Array.from({length: (rows}, (_, i) => i).length)) {
      for (j in Array.from({length: (cols}, (_, i) => i).length)) {
          if (mat[i][j] == rows[i] == cols[j] == 1) {
              result += 1
  return result;
}

export { numSpecial };
